### ISSUE TITLE
unnecessary `unsafe` block when it's needed (cxx related)

### ISSUE THREAD

**Issue by Geobert** [rust-lang/rust-clippy#7552]
Lint name: unused_unsafe

I tried this code:

```rust
#[cxx::bridge]
mod ffi {
  unsafe extern "C++" {
    fn my_fct();
  }
 }

fn out_fct() {
  unsafe { my_fct(); }
} 
```

I expected to see this happen: Compiler says I needed to add `unsafe`, no warning from Clippy

Instead, this happened: Compiler says I needed to add `unsafe`, clippy is emitting a warning about uneeded unsafe

### Meta

- `cargo clippy -V`: clippy 0.1.54 (a178d03 2021-07-26)
- `rustc -Vv`:
```
rustc 1.54.0 (a178d0322 2021-07-26)
binary: rustc
commit-hash: a178d0322ce20e33eac124758e837cbd80a6f633
commit-date: 2021-07-26
host: x86_64-unknown-linux-gnu
release: 1.54.0
LLVM version: 12.0.1
```

**Comment by LeSeulArtichaut** [comment]
The `unused_unsafe` lint is emitted by rustc, not clippy. If there is an issue it should be filed to the [rust-lang/rust](https://github.com/rust-lang/rust) repo.

I'm not able to reproduce your issue however:

```rust
#[cxx::bridge]
mod ffi {
  unsafe extern "C++" {
    fn my_fct();
  }
 }

fn out_fct() {
  my_fct();
}
```
just compiles for me. The reason for this is that CXX generates the following code:

```rust
mod ffi {
    pub fn my_fct() {
        extern "C" {
            #[link_name = "cxxbridge1$my_fct"]
            fn __my_fct();
        }
        unsafe { __my_fct() }
    }
}
```
so `ffi::my_fct` is not an `unsafe` function. Am I missing something?

**Comment by Geobert** [comment]
My bad, rust-analyzer first squiggle it, I though it was the compiler. (Love your nickname btw x) )
